Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -2,285 +2,69 @@
-    }
-
-    public func getInt32() -> Int32 {
-        let result = from(Array(array[currentIndex..<currentIndex + MemoryLayout<Int32>.size]), Int32.self)
-        currentIndex += MemoryLayout<Int32>.size
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getInt32(_ index: Int) -> Int32 {
-        let result = from(Array(array[index..<index + MemoryLayout<Int32>.size]), Int32.self)
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getInt64() -> Int64 {
-        let result = from(Array(array[currentIndex..<currentIndex + MemoryLayout<Int64>.size]), Int64.self)
-        currentIndex += MemoryLayout<Int64>.size
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getInt64(_ index: Int) -> Int64 {
-        let result = from(Array(array[index..<index + MemoryLayout<Int64>.size]), Int64.self)
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getInt() -> Int {
-        let result = from(Array(array[currentIndex..<currentIndex + MemoryLayout<Int>.size]), Int.self)
-        currentIndex += MemoryLayout<Int>.size
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getInt(_ index: Int) -> Int {
-        let result = from(Array(array[index..<index + MemoryLayout<Int>.size]), Int.self)
-        return currentEndianness == .little ? result.littleEndian : result.bigEndian
-    }
-
-    public func getFloat() -> Float {
-        let result = from(Array(array[currentIndex..<currentIndex + MemoryLayout<UInt32>.size]), UInt32.self)
-        currentIndex += MemoryLayout<UInt32>.size
-        return currentEndianness == .little ? Float(bitPattern: result.littleEndian)
-            : Float(bitPattern: result.bigEndian)
-    }
-
-    public func getFloat(_ index: Int) -> Float {
-        let result = from(Array(array[index..<index + MemoryLayout<UInt32>.size]), UInt32.self)
-        return currentEndianness == .little ? Float(bitPattern: result.littleEndian)
-            : Float(bitPattern: result.bigEndian)
-    }
-
-    public func getDouble() -> Double {
-        let result = from(Array(array[currentIndex..<currentIndex + MemoryLayout<UInt64>.size]), UInt64.self)
-        currentIndex += MemoryLayout<UInt64>.size
-        return currentEndianness == .little ? Double(bitPattern: result.littleEndian)
-            : Double(bitPattern: result.bigEndian)
-    }
-
-    public func getDouble(_ index: Int) -> Double {
-        let result = from(Array(array[index..<index + MemoryLayout<UInt64>.size]), UInt64.self)
-        return currentEndianness == .little ? Double(bitPattern: result.littleEndian)
-            : Double(bitPattern: result.bigEndian)
-    }
-
-    public func toByteArray() -> [UInt8] {
-        return array
+        data.append(value)
     }
```

Comment:
I have added a function getData, and all the simple wrapper functions of Data functionality can be executed directly if needed.